src/components/Feedback.js

```diff
@@ -0,0 +1,91 @@
+import { h } from 'preact'
+import { useState, useRef, useEffect } from 'preact/hooks'
+import { useI18n, useSoftkey, useNavigation } from 'hooks'
+import { sendFeedback } from 'utils'
+
+export const Feedback = ({ close }) => {
+  const containerRef = useRef()
+  const i18n = useI18n()
+  const [message, setMessage] = useState()
+  const [validInput, setValidInput] = useState(null)
```

Comment:
Can you explain the meaning of `validInput`? What does it mean when it's `true`, `false`, or `null`.

Reply:
perhaps

`true` - (default value) when user type something even delete them.
`false` - only when the message is null after press send

in this case, then I think we don't need null, 
then we can simplify a lot of the Conditional operator (?:) code in the markup.

Reply:
Hmm perhaps there's a way to simplify things here, let's hash it out. 

> Can you explain the meaning of validInput? What does it mean when it's true, false, or null.

`validInput` is a simple state variable I'm using to keep track of when to display the `error` and `success` messages when users attempts to send feedback. As it stands right now: 

* `null` means neither `success` nor `error` messages are displayed, this is the original state when first rendered
* `true` means the input was valid (`message` wasn't an empty string) and therefore a `success` message is shown
* `false` means the input was invalid (empty string) and therefore the `error` message is shown

I keep track of any change of `validInput` in the dependency arrays of both `useEffect` hooks, and make use of `setValidInput` to make the messages appear and disappear according to the behavior I understood to aim for: user attempts to send with no text, `error` message appears, on typing something error message disappears, the attempt to send again with text this time, `success` message appears for 4 seconds only, etc. 

So if we try to simplify this to just `true` and `false` (no `null`), the issue I'm seeing is what to check for to know when to display `success` message:

```
{validInput === true ? <div class='success'><span>{i18n('feedback-success')}</span></div> : null}
```

Makes sense? Happy to see a simpler way, I'm realizing now maybe the name `validInput` is confusing

Reply:
I think this is confusing because the UX is confusing. 

1. I don't think we should let the users select RSK (send) before there's any text in the textarea. We can easily handle that by looking at `message`. I find the "error" message completely unnecessary.

2. On success, we should direct people away from the form. Filling a form successfully never leads to the same empty form (unless it's a repetitive task by nature). It should lead to a confirmation and then to wherever the user was before (settings menu?). I suggest showing the confirmation message in our regular popup, on top of the feedback form, and going back to /settings when the user OKs the popup.

Reply:
I gotta agree with you, what you're proposing makes sense

Reply:
I agree too! That's better in UX! 

one suggestion here if popup shows the confirmation before sending, that would be even better!

Reply:
> I think this is confusing because the UX is confusing.
> 
> 1. I don't think we should let the users select RSK (send) before there's any text in the textarea. We can easily handle that by looking at `message`. I find the "error" message completely unnecessary.
> 2. On success, we should direct people away from the form. Filling a form successfully never leads to the same empty form (unless it's a repetitive task by nature). It should lead to a confirmation and then to wherever the user was before (settings menu?). I suggest showing the confirmation message in our regular popup, on top of the feedback form, and going back to /settings when the user OKs the popup.

I completely agree that! from a user point of view if the same empty form is displayed again it causes a confusion from the user's point of view!

Reply:
Updated success flow now according to feedback and design changes